jre_emul/android/platform/libcore/ojluni/src/main/java/javax/security/cert/Certificate.java

```diff
@@ -1,79 +1,97 @@
+ * a public key is that of another principal.  (A principal represents
+ * an entity such as an individual user, a group, or a corporation.)
+ *<p>
+ * This class is an abstraction for certificates that have different
+ * formats but important common uses.  For example, different types of
+ * certificates, such as X.509 and PGP, share general certificate
+ * functionality (like encoding and verifying) and
+ * some types of information (like a public key).
  * <p>
- * Note: This package is provided only for compatibility reasons.
- * It contains a simplified version of the java.security.cert package that was
- * previously used by JSSE (Java SSL package). All applications that do not have
- * to be compatible with older versions of JSSE (that is before Java SDK 1.5)
- * should only use java.security.cert.
+ * X.509, PGP, and SDSI certificates can all be implemented by
+ * subclassing the Certificate class, even though they contain different
+ * sets of information, and they store and retrieve the information in
+ * different ways.
+ *
+ * <p><em>Note: The classes in the package {@code javax.security.cert}
+ * exist for compatibility with earlier versions of the
+ * Java Secure Sockets Extension (JSSE). New applications should instead
+ * use the standard Java SE certificate classes located in
+ * {@code java.security.cert}.</em></p>
+ *
+ * @since 1.4
+ * @see X509Certificate
+ *
+ * @author Hemma Prafullchandra
  */
 public abstract class Certificate {
 
     /**
-     * Creates a new {@code Certificate}.
-     */
-    public Certificate() {}
-
-    /**
-     * Compares the argument to this Certificate. If both have the same bytes
-     * they are assumed to be equal.
+     * Compares this certificate for equality with the specified
+     * object. If the {@code other} object is an
+     * {@code instanceof} {@code Certificate}, then
+     * its encoded form is retrieved and compared with the
+     * encoded form of this certificate.
      *
-     * @param obj
-     *            the {@code Certificate} to compare with this object
-     * @return <code>true</code> if {@code obj} is the same as this
-     *         {@code Certificate}, <code>false</code> otherwise
-     * @see #hashCode
+     * @param other the object to test for equality with this certificate.
+     * @return true if the encoded forms of the two certificates
+     *         match, false otherwise.
      */
-    public boolean equals(Object obj) {
-        if (obj == this) {
+    public boolean equals(Object other) {
+        if (this == other)
             return true;
-        }
-        if (!(obj instanceof Certificate)) {
+        if (!(other instanceof Certificate))
             return false;
-        }
-        Certificate object = (Certificate) obj;
+        Certificate otherCert = (Certificate) other;
         try {
-            return Arrays.equals(getEncoded(), object.getEncoded());
+            return Arrays.equals(getEncoded(), otherCert.getEncoded());
```

Comment:
Add comment that this was changed for j2objc. We forgot to do so when we first made that fix, and I don't want it forgotten in future merges.

Reply:
done